### PR TITLE
feat(experimental): `Appearance`, `Icons` change to host directives

### PR DESCRIPTION
**Pull request by waterplea**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #

### CODE REVIEW

File: projects/core/styles/theme/appearance/outline.less [hunk 1]
```diff
@@ -2,31 +2,40 @@
 
 [tuiAppearance][data-appearance='outline'],
 [tuiAppearance][data-appearance='whiteblock'] {
-    background: transparent;
+    --t-bs: var(--tui-base-04);
```
Comment (splincode) [inline projects/core/styles/theme/appearance/outline.less:5]: why not `--t-box-shadow`?

File: projects/addon-mobile/styles/android/checkbox.less [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+    --t-size: 1.125rem;
+
     color: var(--tui-base-01);
     border-radius: 0.125rem;
-    box-shadow: inset 0 0 0 0.125rem var(--tui-base-04);
+    outline: 0.125rem solid var(--tui-base-04);
+    outline-offset: -0.125rem;
+    transition: none;
+    box-shadow: none !important;
+    background: var(--t-background) !important;
```
Comment (vladimirpotekhin) [inline projects/addon-mobile/styles/android/checkbox.less:10]: For what cases "!important" is necessary? I tried to remove it in https://taiga-ui--pr6226-host-dir-gqwo4bsw.web.app/experimental/radio and didn't notice changes
Снимок экрана 2023-12-14 в 10 21 02

File: projects/experimental/directives/icons/icons.directive.ts [hunk 3]
```diff
@@ -30,3 +29,8 @@ export class TuiIconsDirective {
         directiveStyles.addComponent(TuiIconsComponent);
     }
 }
+
+export const TUI_ICONS = {
```
Comment (vladimirpotekhin) [inline projects/experimental/directives/icons/icons.directive.ts:33]: will this work for our users?

**Review by vladimirpotekhin — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-mobile/styles/android/checkbox.less (hunk 2 above)
Comment (waterplea) [reply projects/addon-mobile/styles/android/checkbox.less:10]: Hover effect is shown in without it when it shouldn't

**Pull request merged**